src/input.c

```diff
@@ -38,3 +40,228 @@ char* ReadUserInput(const char* const message) {
 
   return input;
 }
+
+int GetInputInteger(const char* message, int max_value) {
+  bool valid_answer = false;
+  int output = 0;
+  while (!valid_answer) {
+    char* attribute = ReadUserInput(message);
+
+    if (IsInteger(attribute)) {
+      long int attribute_num = strtol(attribute, NULL, kBaseTen);
+
+      if (0 <= attribute_num && attribute_num <= max_value) {
+        output = (int)attribute_num;
+        valid_answer = true;
+      } else {
+        printf("Invalid answer please pick a number between 0 and %d! \n",
+               max_value);
+      }
+    }
+    free(attribute);
+  }
+
+  return output;
+}
+
+bool GetInputBoolean(const char* message) {
+  while (1) {
+    char* attribute_bool = ReadUserInput(message);
+
+    char attribute_bool_answer = attribute_bool[0];
+    free(attribute_bool);
+
+    switch (attribute_bool_answer) {
+      case kYes:
+        return true;
+
+      case kNo:
+        return false;
+
+      default:
+        printf("Invalid answer!\n");
+        break;
+    }
+  }
+}
+
+UserPreferences ManualAttributes(void) {
+  UserPreferences attributes;
+  const int kMaxNum = 10;
+  const int kMaxCarEfficiency = 500;
+  const double kConvertToAttributeDouble = 10.0;
+
+  printf(
+      "To input your personal preferences please write a number between "
+      "0-10!\n");
+
+  double attribute_price =
+      GetInputInteger("Price preference (0-10):", kMaxNum) /
+      kConvertToAttributeDouble;
+  double attribute_time = GetInputInteger("Time preference (0-10):", kMaxNum) /
+                          kConvertToAttributeDouble;
+  double attribute_environment =
+      GetInputInteger("Sustainability preference (0-10):", kMaxNum) /
+      kConvertToAttributeDouble;
+  double attribute_health =
```

Comment:
Im thinking that maybe we should not prompt for health if we do not have calculations for this in the program. Maybe just leave this as 0 out of 10 default and not prompt the user.

Reply:
I agree, but would rather have it in and then disable later if it is not used, but if we are sure it is not used it can be set to zero.